dvc/remote/oss.py

```diff
@@ -0,0 +1,220 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import os
+import re
+import string
+import logging
+
+try:
+    import oss2
+except ImportError:
+    oss2 = None
+
+from dvc.utils import tmp_fname, move
+from dvc.utils.compat import urlparse, makedirs
+from dvc.progress import progress
+from dvc.config import Config
+from dvc.remote.base import RemoteBase
+from dvc.remote.azure import Callback
+
+
+logger = logging.getLogger(__name__)
+
+
+class RemoteOSS(RemoteBase):
+    """
+    oss2 document:
+    https://www.alibabacloud.com/help/doc-detail/32026.htm
+
+
+    Examples
+    ----------
+    $ dvc remote add myremote oss://my-bucket.endpoint/path
+    Set key id and key secret using modify command
+    $ dvc remote modify myremote oss_key_id my-key-id
+    $ dvc remote modify myremote oss_key_secret my-key-secret
+    or environment variables
+    $ export OSS_ACCESS_KEY_ID="my-key-id"
+    $ export OSS_ACCESS_KEY_SECRET="my-key-secret"
+    """
+
+    scheme = "oss"
+    REGEX = (
+        r"^oss://(?P<url>(?P<bucket>.*?)\.(?P<endpoint>.*?))?(?P<path>/.*)?$"
```

Comment:
Let's not mix the endpoint  into the url. Please take a look at s3 driver, it has endpoint configured as a separate value in the remote section. E.g.
```
dvc remote add myremote oss://my-bucket/path
dvc remote modify myremote endpoint my-endpoint
...
```

Reply:
Fixed. Now use the s3 style url for oss storage.